python-package/xgboost/spark/core.py

```diff
@@ -691,7 +684,10 @@ def _fit(self, dataset):
             )
 
         if self._repartition_needed(dataset):
-            dataset = dataset.repartition(num_workers)
+            # Repartition on `monotonically_increasing_id` column to avoid repartition
+            # result unbalance. Directly using `.repartition(N)` might result in some
```

Comment:
May I ask, In which scenario the `data.repartition(num_workers)` might result in some empty partitions?

Reply:
I think it is because of the round robin repartition issue which causes no records shuffled to some partitions

Reply:
Does the "round robin" really cause empty partition? I even can't think of a real case that can cause this issue.

Have you ever encountered the data.repartition(N) to cause the empty partition?

BTW, Does it cause the performance issue here we call repartition twice?

Reply:
I am still debugging...